pkg/container-collection: remove references to gRPC

alban opened the pull request:
# pb.ContainerDefinition has been reduced to the minimum.

After this patch:
```
$ git grep github.com/kinvolk/inspektor-gadget/pkg/gadgettracermanager/api
gadget-container/gadgettracermanager/main.go:   pb "github.com/kinvolk/inspektor-gadget/pkg/gadgettracermanager/api"
gadget-container/hooks/nri/main.go:     pb "github.com/kinvolk/inspektor-gadget/pkg/gadgettracermanager/api"
gadget-container/hooks/oci/main.go:     pb "github.com/kinvolk/inspektor-gadget/pkg/gadgettracermanager/api"
pkg/gadgettracermanager/gadgettracermanager.go: pb "github.com/kinvolk/inspektor-gadget/pkg/gadgettracermanager/api"
```

Before this patch:
```
$ git grep github.com/kinvolk/inspektor-gadget/pkg/gadgettracermanager/api
examples/kube-container-collection/main.go:     pb "github.com/kinvolk/inspektor-gadget/pkg/gadgettracermanager/api"
gadget-container/gadgettracermanager/main.go:   pb "github.com/kinvolk/inspektor-gadget/pkg/gadgettracermanager/api"
gadget-container/hooks/nri/main.go:     pb "github.com/kinvolk/inspektor-gadget/pkg/gadgettracermanager/api"
gadget-container/hooks/oci/main.go:     pb "github.com/kinvolk/inspektor-gadget/pkg/gadgettracermanager/api"
pkg/container-collection/container-collection.go:       pb "github.com/kinvolk/inspektor-gadget/pkg/gadgettracermanager/api"
pkg/container-collection/interface.go:  pb "github.com/kinvolk/inspektor-gadget/pkg/gadgettracermanager/api"
pkg/container-collection/match.go:      pb "github.com/kinvolk/inspektor-gadget/pkg/gadgettracermanager/api"
pkg/container-collection/match_test.go: pb "github.com/kinvolk/inspektor-gadget/pkg/gadgettracermanager/api"
pkg/container-collection/options.go:    pb "github.com/kinvolk/inspektor-gadget/pkg/gadgettracermanager/api"
pkg/controllers/trace_controller.go:    pb "github.com/kinvolk/inspektor-gadget/pkg/gadgettracermanager/api"
pkg/gadgets/dns/gadget.go:      pb "github.com/kinvolk/inspektor-gadget/pkg/gadgettracermanager/api"
pkg/gadgets/helpers.go: pb "github.com/kinvolk/inspektor-gadget/pkg/gadgettracermanager/api"
pkg/gadgets/network-graph/gadget.go:    pb "github.com/kinvolk/inspektor-gadget/pkg/gadgettracermanager/api"
pkg/gadgets/seccomp/gadget.go:  pb "github.com/kinvolk/inspektor-gadget/pkg/gadgettracermanager/api"
pkg/gadgets/snisnoop/gadget.go: pb "github.com/kinvolk/inspektor-gadget/pkg/gadgettracermanager/api"
pkg/gadgettracermanager/containers-map/tracer.go:       pb "github.com/kinvolk/inspektor-gadget/pkg/gadgettracermanager/api"
pkg/gadgettracermanager/gadgettracermanager.go: pb "github.com/kinvolk/inspektor-gadget/pkg/gadgettracermanager/api"
pkg/gadgettracermanager/gadgettracermanager_test.go:    pb "github.com/kinvolk/inspektor-gadget/pkg/gadgettracermanager/api"
pkg/gadgettracermanager/k8s/k8s.go:     pb "github.com/kinvolk/inspektor-gadget/pkg/gadgettracermanager/api"
pkg/gadgettracermanager/pubsub/pubsub.go:       pb "github.com/kinvolk/inspektor-gadget/pkg/gadgettracermanager/api"
pkg/gadgettracermanager/pubsub/pubsub_test.go:  pb "github.com/kinvolk/inspektor-gadget/pkg/gadgettracermanager/api"
pkg/local-gadget-manager/local-gadget-manager.go:       pb "github.com/kinvolk/inspektor-gadget/pkg/gadgettracermanager/api"
pkg/tracer-collection/tracer-collection.go:     pb "github.com/kinvolk/inspektor-gadget/pkg/gadgettracermanager/api"
```

## How to use

No changes

## Testing done

WIP

mauriciovasquezbernal commented on pkg/gadgettracermanager/api/gadgettracermanager.proto:
```diff
@@ -82,33 +60,42 @@ message OwnerReference {
-  string podname = 6;
-  string name = 7;
-  repeated Label labels = 8;
+  //string namespace = 5;
+  //string podname = 6;
+  //string name = 7;
+  //repeated Label labels = 8;
```
Should all these commented members be removed?

mauriciovasquezbernal reviewed (commented):
It's looking great!

alban commented:
I moved the following packages in container-collection:
- pkg/gadgettracermanager/k8s
- pkg/gadgettracermanager/pubsub

I removed AddTracer/RemoveTracer from the gRPC API because:
- They were not used anymore
- Keeping them would require to write unnecessary code to convert `pb.ContainerSelector` into `containercollection.ContainerSelector`.

I needed to rename fields (`Id` -> `ID` etc.) so that the linter pass.

I used the type `OciConfig *ocispec.Spec` since we are no longer constrained by gRPC types, as explained in https://github.com/kinvolk/inspektor-gadget/issues/735.

mauriciovasquezbernal reviewed (approved):
Nice cleanup. LGTM!
